crates/aiken-lang/src/parser/expr/mod.rs

```diff
@@ -40,157 +42,167 @@ pub fn parser(
     sequence: Recursive<'_, Token, UntypedExpr, ParseError>,
 ) -> impl Parser<Token, UntypedExpr, Error = ParseError> + '_ {
     recursive(|expression| {
-        let chained_debugged = chained(sequence, expression)
-            .then(just(Token::Question).or_not())
-            .map_with_span(|(value, token), location| match token {
-                Some(_) => UntypedExpr::TraceIfFalse {
-                    value: Box::new(value),
-                    location,
-                },
-                None => value,
-            });
-
-        // Negate
-        let op = choice((
-            just(Token::Bang).to(ast::UnOp::Not),
-            just(Token::Minus)
-                // NOTE: Prevent conflict with usage for '-' as a standalone binary op.
-                // This will make '-' parse when used as standalone binop in a function call.
-                // For example:
-                //
-                //    foo(a, -, b)
-                //
-                // but it'll fail in a let-binding:
-                //
-                //    let foo = -
-                //
-                // which seems acceptable.
-                .then_ignore(just(Token::Comma).not().rewind())
-                .to(ast::UnOp::Negate),
-        ));
-
-        let unary = op
-            .map_with_span(|op, span| (op, span))
-            .repeated()
-            .then(chained_debugged)
-            .foldr(|(un_op, span), value| UntypedExpr::UnOp {
-                op: un_op,
-                location: span.union(value.location()),
+        choice((
+            error_todo(sequence.clone()),
+            pure_expression(sequence, expression),
+        ))
```

Comment:
Note: the diff is bad in this file but I simply moved the current content of the function into a new `pure_expression` function, and replaced the body of the main parser with this choice parser.